domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1802,59 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// AllModelActivationStatusQuery returns a SQL statement that will return all model activated status.
+func (st *State) AllModelActivationStatusQuery() string {
+	return "SELECT activated from model"
+}
+
+// GetActivatedModelUUIDs returns the UUIDs of all activated models from the given list of UUIDs.
+//
+// Returns:
+//   - []string: A subset of the input UUIDs containing only those of activated models.
+//   - error: An error if the retrieval process fails.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []string) ([]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type modelUUID struct {
+		UUID coremodel.UUID `db:"uuid"`
+	}
+
+	type UUIDs []string
+	inputUUIDs := UUIDs(uuids)
+
+	stmt, err := st.Prepare(`
+SELECT 
+uuid AS &modelUUID.uuid
+FROM   model
+WHERE uuid IN ($UUIDs[:])
+AND activated = true
```

Comment:
If you use `v_model` you don't need the activated check. Only activated model's appear in this view.